packages/nodes-base/nodes/Google/Drive/GoogleDrive.node.ts

```diff
@@ -2031,6 +2036,70 @@ export class GoogleDrive implements INodeType {
 	};
 
 	methods = {
+		credentialTest: {
+			async testGoogleDriveCredentials(
+				this: ICredentialTestFunctions,
+				credential: ICredentialsDecrypted,
+			): Promise<INodeCredentialTestResult> {
+				const {
+					oauthTokenData: { access_token: accessToken },
+				} = credential.data as { oauthTokenData: { access_token: string } };
+
+				try {
+					/*
+						Do a first request to make sure the API is enabled
+						in the Google console
+					*/
+					await this.helpers.request({
+						method: 'GET',
+						uri: 'https://www.googleapis.com/drive/v3/files',
+						headers: {
+							Authorization: `Bearer ${accessToken}`,
+						},
+						json: true,
+					});
+
+					/*
+						Do a second request to get the scopes the user accepted,
+						and compare it to the scopes in the credentials (most up-to-date scopes).
+						If the scopes are not the same, let the user know that some funcionality
+						from the nodes might not work due to the missing scopes.
+					*/
+
+					const { scope } = (await this.helpers.request({
+						method: 'GET',
+						uri: 'https://www.googleapis.com/oauth2/v1/tokeninfo',
+						qs: {
+							access_token: accessToken,
+						},
+						json: true,
+					})) as { scope: string };
+
+					const currentNodeScopes = [
+						'https://www.googleapis.com/auth/drive',
+						'https://www.googleapis.com/auth/drive.appdata',
+						'https://www.googleapis.com/auth/drive.photos.readonly',
```

Comment:
there would be a lot of code duplication in such approach, we should create folder with scopes as constants and then use import in credential and node